src/base/preferences.cpp

```diff
@@ -448,6 +448,26 @@ void Preferences::setActionOnDblClOnTorrentFn(int act)
     setValue("Preferences/Downloads/DblClOnTorFn", act);
 }
 
+bool Preferences::deleteAddedTorrents() const
```

Comment:
>The thing is there is no single function which does the torrent import (and that is bad design). 

I agree with you. I don't like it. But, in any case, now it is much better than it was before 3.3.x. It's a pretty complex case. I'm considering it a little from time to time. And I think I found some solution on this problem. If I'm not mistaken, I will publish it later.

>Hence the class to reuse the code. The preferences are in the TorrentFileProcessignGuard now.

Okay. At least it's good as a workaround.